common/include/scipp/common/except.h

```diff
@@ -5,8 +5,12 @@
 #ifndef SCIPP_COMMON_EXCEPT_H
 #define SCIPP_COMMON_EXCEPT_H
 
+#include <initializer_list>
+#include <sstream>
```

Comment:
Why is this required? I think it is quite a large header that we should avoid including everywhere.